### ISSUE TITLE
hi

### ISSUE THREAD

**Issue by mohsen202** [alejandro-ao/langchain-ask-pdf#27]
hi. please help me. how to create custom model from many pdfs in Persian language? tank you.

**Comment by SanjeevDeori** [comment]
To create a custom model from many PDFs in Persian, you need to follow these steps:

1. Extract Text from PDFs

Use PyMuPDF (fitz), pdfplumber, or pdfminer.six to extract text from PDFs.

Example using PyMuPDF:

import fitz

def extract_text_from_pdf(pdf_path):
    doc = fitz.open(pdf_path)
    text = ""
    for page in doc:
        text += page.get_text("text") + "\n"
    return text

text = extract_text_from_pdf("sample.pdf")
print(text)


2. Preprocess the Text

Persian text may need normalization using libraries like hazm or parsivar.

Example with hazm:

from hazm import Normalizer

normalizer = Normalizer()
text = normalizer.normalize(text)


3. Tokenization & Cleaning

Use hazm for tokenization:

from hazm import word_tokenize

tokens = word_tokenize(text)
print(tokens)


4. Train a Model

If you want a text classification model, use sklearn or transformers.

If you want a language model, fine-tune a Persian NLP model like bert-base-parsbert-uncased.


Example: Fine-tune a Transformer Model (Hugging Face)

from transformers import AutoTokenizer, AutoModelForSequenceClassification

model_name = "HooshvareLab/bert-fa-base-uncased"
tokenizer = AutoTokenizer.from_pretrained(model_name)
model = AutoModelForSequenceClassification.from_pretrained(model_name, num_labels=2)

5. Train the Model with Your Data

Convert extracted text into labeled datasets.

Use transformers or sklearn for training.
